server/service/src/processors/transfer/requisition/mod.rs

```diff
@@ -0,0 +1,190 @@
+pub(crate) mod create_response_requisition;
+pub(crate) mod link_request_requisition;
+pub(crate) mod update_request_requisition_status;
+
+#[cfg(test)]
+pub(crate) mod test;
+
+use repository::{
+    ChangelogAction, ChangelogFilter, ChangelogRepository, ChangelogRow, ChangelogTableName,
+    EqualFilter, KeyValueStoreRepository, KeyValueType, RepositoryError, Requisition,
+    RequisitionFilter, RequisitionRepository, StorageConnection,
+};
+use thiserror::Error;
+
+use crate::{
+    processors::transfer::requisition::{
+        create_response_requisition::CreateResponseRequisitionProcessor,
+        link_request_requisition::LinkeRequestRequisitionProcessor,
+        update_request_requisition_status::UpdateRequestRequstionStatusProcessor,
+    },
+    service_provider::ServiceProvider,
+    sync::{ActiveStoresOnSite, GetActiveStoresOnSiteError},
+};
+
+const CHANGELOG_BATCH_SIZE: u32 = 20;
+
+#[derive(Clone, Debug)]
+pub(crate) struct RequisitionTransferProcessorRecord {
+    requisition: Requisition,
+    linked_requisition: Option<Requisition>,
+    other_party_store_id: String,
+}
+
+#[derive(Error, Debug)]
+pub(crate) enum ProcessRequisitionTransfersError {
+    #[error("Problem getting upsert record {0}")]
+    GetUpsertRecordError(GetUpsertRecordError),
+    #[error("{0}")]
+    GetActiveStoresOnSiteError(GetActiveStoresOnSiteError),
+    #[error("{0:?}")]
+    DatabaseError(RepositoryError),
+    #[error("{0}")]
+    ProcessorError(ProcessorError),
+    #[error("Name id is missing from requisition changelog {0:?}")]
+    NameIdIsMissingFromChangelog(ChangelogRow),
+    #[error("Name is not an active store {0:?}")]
+    NameIsNotAnAciveStore(ChangelogRow),
+}
+
+// TODO transaction
+
+pub(crate) fn process_requisition_transfers(
+    service_provider: &ServiceProvider,
+) -> Result<(), ProcessRequisitionTransfersError> {
+    use ProcessRequisitionTransfersError as Error;
+    let processors: Vec<Box<dyn RequisitionTransferProcessor>> = vec![
+        Box::new(CreateResponseRequisitionProcessor),
+        Box::new(LinkeRequestRequisitionProcessor),
+        Box::new(UpdateRequestRequstionStatusProcessor),
+    ];
+
+    let ctx = service_provider.context().map_err(Error::DatabaseError)?;
+
+    let active_stores =
```

Comment:
Edge case I thought of here, is if an active store gets created during processing. I assume that's ok to ignore?

Reply:
Super edge case I think, it's possible though, I think risk is super low and so is the impact, but thanks for thinking of an edge case. I think it's ok to not deal with this edge case

Reply:
> Can sync start while the processors are running from a previous sync

Currently yeah it's possible, triggers for processors do not await for processing to finish. Added to [improvements](https://github.com/openmsupply/open-msupply/issues/608)

> I suspect this is also think in edge case with low risk bucket too, just thinking.

Yeah I think it's quite an edge case

> this queues to siteA all the records owned by storeB, as well as the records where storeB is the other party)

Ouch, i was going to suggest adding sanity check on remote site which doesn't allow owned record to be synced to remote site after initialisation, above sounds like it would re-sync siteA owned records to siteA (where storeB is other party) ?

Reply:
>Ouch, i was going to suggest adding sanity check on remote site which doesn't allow owned record to be synced to remote site after initialisation, above sounds like it would re-sync siteA owned records to siteA (where storeB is other party) ?

Yes I believe that's what'll happen, as the "other party" records relative to storeB are records owned by siteA. It should be OK as the remote does push before pull, it's pretty guaranteed the central server should only queue data that is the latest on the remote anyway (unless central broke the rules and edited the record).

We could add logic to prevent this on the v5 API (only queue "other party" records where the owner of the records is not another store on the destination site being intialised), potentially adds a fair bit of complexity though.